src/wagtail_vector_index/backends/numpy/backend.py

```diff
@@ -37,8 +37,8 @@ def similarity_search(
         sorted_similarities = sorted(
             similarities, key=lambda pair: pair[0], reverse=True
         )
-        for similarity in [pair[1] for pair in sorted_similarities][:limit]:
-            yield SearchResponseDocument(id=similarity.id, metadata=similarity.metadata)
+        for document in [pair[1] for pair in sorted_similarities][:limit]:
```

Comment:
**Nitpick (to address as a separate PR):** Wouldn't it be more efficient to have one for loop?